pkg/patterns/addon/patch.go

```diff
@@ -35,25 +35,38 @@ import (
 func ApplyPatches(ctx context.Context, object declarative.DeclarativeObject, objects *manifest.Objects) error {
 	log := log.Log
 
-	p, ok := object.(addonsv1alpha1.Patchable)
-	if !ok {
-		return fmt.Errorf("provided object (%T) does not implement Patchable type", object)
-	}
-
+	var p addonsv1alpha1.Patchable
 	var patches []*unstructured.Unstructured
 
-	for _, p := range p.PatchSpec().Patches {
-		// Object is nil, Raw  is populated (with json, even when input was yaml)
-		r := bytes.NewReader(p.Raw)
-		decoder := yaml.NewYAMLOrJSONDecoder(r, 1024)
-		patch := &unstructured.Unstructured{}
+	unstruct, ok := object.(*unstructured.Unstructured)
+	if ok {
+		patch, _, err := unstructured.NestedSlice(unstruct.Object, "spec", "patches")
+		if err != nil {
+			return fmt.Errorf("unable to get patches from unstruct: %v", err)
```

Comment:
Nit: unstructured, not unstruct, as we expect this will eventually bubble-up to the user.